### PR TITLE
feat(template): Adding pydocstyle to the template

### PR DESCRIPTION
**Pull request by YurelyCamacho**
## Pull Request description
Adding pydocstyle as an option to the template and setting pre-commit hook

Solving issue #37

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [x ] new feature
- [ ] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x ] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [x ] New and old tests passed locally.

## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [x ] I managed to test the new changes locally
- [ x] I confirm that the issues mentioned were fixed/resolved .
```

### CODE REVIEW

**Comment by YurelyCamacho** [comment]
@xmnlab I noticed that in Anavelyz's PR on bandit, you asked him to add some documentation. I'm going to do the same here.

File: {{cookiecutter.project_slug}}/pyproject.toml [hunk 1]
```diff
@@ -90,7 +90,10 @@ myst-parser = "^0.18.1"
 {%- elif cookiecutter.documentation_engine == 'jupyter-book' %}
 jupyter-book = "^0.12.3"
 myst-parser = "^0.15.0"
-{%+ endif %}
+{%- endif %}
+{%- if cookiecutter.use_pydocstyle == 'yes' +%}
+pydocstyle = "6.3.0"
```
Comment (xmnlab) [inline {{cookiecutter.project_slug}}/pyproject.toml:95]:
```suggestion
pydocstyle = "^6.3.0"
```

**Review by xmnlab — CHANGES_REQUESTED** [review]
As you mentioned we need the docs in the project and the template side.
Other than that, just added a small comment here about the pkg  version

**Comment by YurelyCamacho** [comment]
@xmnlab Sorry, I don't know what I did here. Could you help me, please

**Comment by xmnlab** [comment]
No worries, just open a new PR with the missing pieces

**Comment by YurelyCamacho** [comment]
@xmnlab Everything is already in this PR, if you can check it, please. The last thing I added was the documentation and the change in version.

**Comment by xmnlab** [comment]
@YurelyCamacho this PR is already merged
the commit you pushed after you merged the PR will not be included to the main anymore.
please open a new PR with the latest changes. thanks!

**Comment by xmnlab** [comment]
although, it seems your last commit is just a revert, is that correct?
sorry, I am a bit confused.

File: {{cookiecutter.project_slug}}/docs/index.md [hunk 2]
```diff
@@ -12,6 +12,9 @@
 
 ## Features
 
+- This package uses [pydocstyle](http://www.pydocstyle.org/en/stable/)
```
Comment (xmnlab) [inline {{cookiecutter.project_slug}}/docs/index.md:15]: you will need to check first if the user selected pydocstyle.
do it in a new PR.

File: {{cookiecutter.project_slug}}/README.md [hunk 3]
```diff
@@ -10,6 +10,9 @@
 
 ## Features
 
+- This package uses [pydocstyle](http://www.pydocstyle.org/en/stable/)
```
Comment (xmnlab) [inline {{cookiecutter.project_slug}}/README.md:13]: you will need to check first if the user selected pydocstyle.
do it in a new PR.

**Pull request merged**